File: dairytrack-web/src/api/peternakan/blog.js
import { fetchAPI } from "../apiClient";

// GET semua data blog
export const getBlogs = () => fetchAPI("blogs");

// GET satu blog by ID
export const getBlogById = (id) => fetchAPI(`blogs/${id}`);

// CREATE blog baru
export const createBlog = async (formData) => {
  try {
    const response = await fetchAPI("blogs", "POST", formData, true); // true untuk FormData

    if (!response.ok) {
      const contentType = response.headers.get("content-type");
      if (contentType && contentType.includes("application/json")) {
        const errorData = await response.json();
        throw new Error(errorData.message || "Terjadi kesalahan pada server.");
      } else {
        const errorText = await response.text();
        throw new Error(errorText || "Terjadi kesalahan pada server.");
      }
    }

    const contentType = response.headers.get("content-type");
    if (contentType && contentType.includes("application/json")) {
      return await response.json();
    } else {
      throw new Error("Respons dari server bukan JSON.");
    }
  } catch (error) {
    console.error("Error in createBlog:", error.message);
    throw error;
  }
};
// UPDATE blog
export const updateBlog = async (id, formData) => {
  try {
    const response = await fetchAPI(`blogs/${id}`, "PUT", formData, true); // true untuk FormData

    if (!response.ok) {
      const contentType = response.headers.get("content-type");
      if (contentType && contentType.includes("application/json")) {
        const errorData = await response.json();
        throw new Error(errorData.message || "Terjadi kesalahan pada server.");
      } else {
        const errorText = await response.text();
        throw new Error(errorText || "Terjadi kesalahan pada server.");
      }
    }

    const contentType = response.headers.get("content-type");
    if (contentType && contentType.includes("application/json")) {
      return await response.json();
    } else {
      throw new Error("Respons dari server bukan JSON.");
    }
  } catch (error) {
    console.error("Error in updateBlog:", error.message);
    throw error;
  }
};
// DELETE blog
export const deleteBlog = (id) => fetchAPI(`blogs/${id}`, "DELETE");

// GET blog photo by ID
export const getBlogPhoto = (id) => fetchAPI(`blogs/${id}/photo`);

File: dairytrack-mobile/lib/config/api/peternakan/blog.dart
import 'package:dairy_track/config/configApi5000.dart';

// GET semua data blog
Future<List<dynamic>> getBlogs() async {
  final response = await fetchAPI("blogs");
  if (response['status'] == 200) {
    return response['data'];
  } else {
    throw Exception(response['message']);
  }
}

// GET satu blog by ID
Future<Map<String, dynamic>> getBlogById(String id) async {
  final response = await fetchAPI("blogs/$id");
  if (response['status'] == 200) {
    return response['data'];
  } else {
    throw Exception(response['message']);
  }
}

// CREATE blog baru
Future<Map<String, dynamic>> createBlog(Map<String, dynamic> formData) async {
  final response = await fetchAPI(
    "blogs",
    method: "POST",
    data: formData,
    isFormData: true,
  );
  if (response['status'] == 201) {
    return response['data'];
  } else {
    throw Exception(response['message']);
  }
}

// UPDATE blog
Future<Map<String, dynamic>> updateBlog(
    String id, Map<String, dynamic> formData) async {
  final response = await fetchAPI(
    "blogs/$id",
    method: "PUT",
    data: formData,
    isFormData: true,
  );
  if (response['status'] == 200) {
    return response['data'];
  } else {
    throw Exception(response['message']);
  }
}

// DELETE blog
Future<void> deleteBlog(String id) async {
  final response = await fetchAPI("blogs/$id", method: "DELETE");
  if (response['status'] != 204) {
    throw Exception(response['message']);
  }
}

// GET blog photo by ID
Future<String> getBlogPhoto(String id) async {
  final response = await fetchAPI("blogs/$id/photo");
  if (response['status'] == 200) {
    return response['data']['photoUrl'];
  } else {
    throw Exception(response['message']);
  }
}
